src/Mod/PartDesign/Gui/TaskPipeParameters.cpp

```diff
@@ -288,6 +305,30 @@ void TaskPipeParameters::removeFromListWidget(QListWidget* widget, QString items
     }
 }
 
+void TaskPipeParameters::onDeleteEdge()
+{
+    // Delete the selected path edge
+    int row = ui->listWidgetReferences->currentRow();
+    QListWidgetItem* item = ui->listWidgetReferences->item(row);
+    if (item) {
+        QByteArray data = item->data(Qt::UserRole).toByteArray();
+        ui->listWidgetReferences->takeItem(row);
+        delete item;
+        // search inside the list of spines
+        PartDesign::Pipe* pipe = static_cast<PartDesign::Pipe*>(vp->getObject());
+        std::vector<std::string> refs = pipe->Spine.getSubValues();
+        std::string obj = data.toStdString();

```

Comment:
‘class QByteArray’ has no member named ‘toStdString'

This doesn't exist in Qt4 so it fails.

Reply:
> This doesn't exist in Qt4 so it fails.

Why do we care about Qt 4? These Qt versions are not supported in terms of security fixes since years so it is a security risk to use them. With Qt 5 (released almost 8 years ago) class QByteArray’ has a member named ‘toStdString'.
Please update your Qt because there is no reason to use 8 year old and since 5 years unmaintained software.